hw/drivers/sensors/mpu6050/src/mpu6050.c

```diff
@@ -0,0 +1,466 @@
+#endif
+
+/* Exports for the sensor API */
+static int mpu6050_sensor_read(struct sensor *, sensor_type_t,
+        sensor_data_func_t, void *, uint32_t);
+static int mpu6050_sensor_get_config(struct sensor *, sensor_type_t,
+        struct sensor_cfg *);
+
+static const struct sensor_driver g_mpu6050_sensor_driver = {
+    mpu6050_sensor_read,
+    mpu6050_sensor_get_config
+};
+
+/**
+ * Writes a single byte to the specified register
+ *
+ * @param The sensor interface
+ * @param The I2C address to use
+ * @param The register address to write to
+ * @param The value to write
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+mpu6050_write8(struct sensor_itf *itf, uint8_t addr, uint8_t reg,
+                  uint32_t value)
+{
+    int rc;
+    uint8_t payload[2] = { reg, value & 0xFF };
+
+    struct hal_i2c_master_data data_struct = {
+        .address = addr,
+        .len = 2,
+        .buffer = payload
+    };
+
+    rc = hal_i2c_master_write(itf->si_num, &data_struct,
+                              OS_TICKS_PER_SEC / 10, 1);
+
+    if (rc != 0) {
+        MPU6050_ERR("Failed to write to 0x%02X:0x%02X with value 0x%02X\n",
+                       addr, reg, value);
+#if MYNEWT_VAL(MPU6050_STATS)
+        STATS_INC(g_mpu6050stats, read_errors);
+#endif
+    }
+
+    return rc;
+}
+
+/**
+ * Reads a single byte from the specified register
+ *
+ * @param The sensor interface
+ * @param The I2C address to use
+ * @param The register address to read from
+ * @param Pointer to where the register value should be written
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+mpu6050_read8(struct sensor_itf *itf, uint8_t addr, uint8_t reg,
+                 uint8_t *value)
+{
+    int rc;
+
+    struct hal_i2c_master_data data_struct = {
+        .address = addr,
+        .len = 1,
+        .buffer = &reg
+    };
+
+    /* Register write */
+    rc = hal_i2c_master_write(itf->si_num, &data_struct,
+                              OS_TICKS_PER_SEC / 10, 0);
+    if (rc != 0) {
+        MPU6050_ERR("I2C access failed at address 0x%02X\n", addr);
+#if MYNEWT_VAL(MPU6050_STATS)
+        STATS_INC(g_mpu6050stats, write_errors);
+#endif
+        return rc;
+    }
+
+    /* Read one byte back */
+    data_struct.buffer = value;
+    rc = hal_i2c_master_read(itf->si_num, &data_struct,
+                             OS_TICKS_PER_SEC / 10, 1);
+
+    if (rc != 0) {
+         MPU6050_ERR("Failed to read from 0x%02X:0x%02X\n", addr, reg);
+ #if MYNEWT_VAL(MPU6050_STATS)
+         STATS_INC(g_mpu6050stats, read_errors);
+ #endif
+    }
+    return rc;
+}
+
+/**
+ * Reads a six bytes from the specified register
+ *
+ * @param The sensor interface
+ * @param The I2C address to use
+ * @param The register address to read from
+ * @param Pointer to where the register value should be written
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+mpu6050_read48(struct sensor_itf *itf, uint8_t addr, uint8_t reg,
```

Comment:
The device I2C interface address is already present in sensor_itf as "si_addr". It generally gets set in the bsp for onboard sensors and in the creator for offboard ones. The parameter "addr" is not needed. You already set the interface address. All you would have to do here is use the macro "SENSOR_GET_ITF()" to get the interface and then use "si_addr" field.